Assignment9/WpfApp/WpfApp/MainWindowViewModel.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Threading;
+
+namespace WpfApp
+{
+    public class MainWindowViewModel : INotifyPropertyChanged
+    {
+        public event PropertyChangedEventHandler? PropertyChanged;
+        public GenericCommand NewContactCommand { get; }
+
+        public GenericCommand RemoveContactCommand { get; }
+
+        public GenericCommand EditContactCommand { get; }
+
+        private ObservableCollection<ContactViewModel> _Contacts = new();
+        public ObservableCollection<ContactViewModel> Contacts
+        {
+            get => _Contacts;
+            set => SetProperty(ref _Contacts, value);
+        }
+
+
+        private ContactViewModel? _SelectedContact;
+        public ContactViewModel SelectedContact
+        {
+            get => _SelectedContact!;
+            set
+            {
+                if (SetProperty(ref _SelectedContact, value))
+                    IsEditContact = false;
+            }
+        }
+
+        private bool _IsEditContact;
+        public bool IsEditContact
+        {
+            get => _IsEditContact;
+            set
+            {
+                if (SetProperty(ref _IsEditContact, value))
+                {
+                    UpdateEditText();
+                    UpdateListBox();
+                }
+
+            }
+        }
+
+        private string _EditSaveText = "Edit";
+        public string EditSaveText
+        {
+            get => _EditSaveText;
+            set => SetProperty(ref _EditSaveText, value);
+        }
+
+        private void UpdateListBox()
```

Comment:
nit: In general I try to keep UI elements out of my view models (even by name)
```suggestion
        private void UpdateContactsList()
```